src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -3,25 +3,26 @@ package mesosphere.marathon.state
 import com.wix.accord.dsl._
 import com.wix.accord._
 import mesosphere.marathon.api.v2.Validation._
-import org.apache.mesos.{ Protos => Mesos }
+
+import org.apache.mesos.Protos.ContainerInfo
+
 import scala.collection.immutable.Seq
 
-// TODO: trait Container and specializations?
-// Current implementation with type defaulting to DOCKER and docker to NONE makes no sense
-case class Container(
-    `type`: Mesos.ContainerInfo.Type = Mesos.ContainerInfo.Type.DOCKER,
-    volumes: Seq[Volume] = Nil,
-    docker: Option[Container.Docker] = None) {
+sealed trait Container {
+  val volumes: Seq[Volume]
 
-  import Container._
+  def docker(): Option[Container.Docker] = {
```

Comment:
Good point! Looks like the containerizer is assumed to be Docker, when the docker field is `Some`. This may lead to problems with the Mesos containerizer, that now also can set this field.